src/AutoRest.CSharp/Mgmt/Generation/ResourceWriter.cs

```diff
@@ -472,7 +472,7 @@ protected override void WriteResourceCollectionEntry(Resource resource)
             _writer.WriteXmlDocumentationReturns($"An object representing collection of {resource.Type.Name.LastWordToPlural()} and their operations over a {_resource.Type.Name}.");
             _writer.WriteXmlDocumentationParameters(collection.ExtraConstructorParameters);
             var extraConstructorParameters = collection.ExtraConstructorParameters;
-            _writer.Append($"public {collection.Type.Name} Get{resource.Type.Name.ResourceNameToPlural()}(");
+            _writer.Append($"public {GetVirtual(true)} {collection.Type.Name} Get{resource.Type.Name.ResourceNameToPlural()}(");
```

Comment:
if `GetVirtual` is always called with `true`, why not just add `virtual` to the string directly?

Reply:
And while you're here, please change `{collection.Type.Name}` to `{collection.Type}`.

Reply:
> if `GetVirtual` is always called with `true`, why not just add `virtual` to the string directly?

Use `GetVirtual` to keep consistency, like [here](https://github.com/Azure/autorest.csharp/blob/9d6a700eea66d61eb5740af40ff7a1cc5e63f1ea/src/AutoRest.CSharp/Mgmt/Generation/ResourceWriter.cs#L317)

Reply:
> And while you're here, please change `{collection.Type.Name}` to `{collection.Type}`.

Changed